shenyu-agent/shenyu-agent-plugins/shenyu-agent-plugin-logging/shenyu-agent-plugin-logging-rocketmq/src/main/java/org/apache/shenyu/agent/plugin/logging/rocketmq/RocketMQLogCollector.java

```diff
@@ -0,0 +1,123 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.shenyu.agent.plugin.logging.rocketmq;
+
+import org.apache.shenyu.agent.plugin.logging.LogCollector;
+import org.apache.shenyu.agent.plugin.logging.spi.LogCollectClient;
+import org.apache.shenyu.common.utils.JsonUtils;
+import org.apache.shenyu.common.utils.ThreadUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.LinkedBlockingDeque;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+/**
+ * queue-based logging collector.
+ */
+public class RocketMQLogCollector implements LogCollector {
+
+    private static final Logger LOG = LoggerFactory.getLogger(RocketMQLogCollector.class);
+    
+    private static final ExecutorService THREAD_EXECUTOR = Executors.newSingleThreadExecutor();
+
+    private static LogCollector instance;
+
+    private final AtomicBoolean started = new AtomicBoolean(false);
+
+    private final LogCollectClient logCollectClient;
+
+    private final BlockingQueue<String> bufferQueue;
+
+    private final int bufferSize = 50000;
+
+    private final int batchSize = 100;
+
+    private final int diffTimeMSForPush = 100;
+
+    private long lastPushTime;
+
+    public RocketMQLogCollector(final LogCollectClient logCollectClient) {
+        this.logCollectClient = logCollectClient;
+        bufferQueue = new LinkedBlockingDeque<>(bufferSize);
+        started.set(true);
+        instance = this;
+        THREAD_EXECUTOR.execute(this::consume);
+    }
+
+    /**
+     * get single instance.
+     *
+     * @return LogCollector is used to collect log
+     */
+    public static LogCollector getInstance() {
+        return instance;
+    }
+
+    /**
+     * batch and async consume.
+     */
+    private void consume() {
+        while (started.get()) {
+            try {
+                List<String> logs = new ArrayList<>();
+                int size = bufferQueue.size();
+                long time = System.currentTimeMillis();
+                long timeDiffMs = time - lastPushTime;
+                if (size >= batchSize || timeDiffMs > diffTimeMSForPush) {
+                    bufferQueue.drainTo(logs, batchSize);
+                    logCollectClient.collect(logs);
+                    lastPushTime = time;
+                } else {
+                    ThreadUtils.sleep(TimeUnit.MILLISECONDS, diffTimeMSForPush);
+                }
+            } catch (Exception e) {
+                LOG.error("DefaultLogCollector collect log error", e);
```

Comment:
just like it logged, i think what this method did is like a  `DefaultLogCollector` should do, maybe it should put in a `DefaultLogCollector`

Reply:
good idea!  I got it.